mlflow/pyfunc/__init__.py

```diff
@@ -201,20 +203,62 @@ def spark_udf(spark, path, run_id=None, result_type="double"):
     Parameters passed to the UDF are forwarded to the model as a DataFrame where the names are
     ordinals (0, 1, ...).
 
+    The predictions are filtered to contain only the columns that can be represented as the
+    ``result_type``. If the ``result_type`` is string or array of strings, all predictions are
+    converted to string. If the result type is not an array type, the left most column with
+    matching type will be returned.
+
     >>> predict = mlflow.pyfunc.spark_udf(spark, "/my/local/model")
     >>> df.withColumn("prediction", predict("name", "age")).show()
 
     :param spark: A SparkSession object.
     :param path: A path containing a :py:mod:`mlflow.pyfunc` model.
     :param run_id: ID of the run that produced this model. If provided, ``run_id`` is used to
                    retrieve the model logged with MLflow.
-    :return: Spark UDF type returned by the model's prediction method. Default double.
+    :param result_type: the return type of the user-defined function. The value can be either a
+                        :class:`pyspark.sql.types.DataType` object or a DDL-formatted type string.
+                        Only a primitive type or an array of primitive types are allowed.
+                        The following classes of result type are supported:
+                        - "int" or instanceof IntegerType: The leftmost integer that can fit in
```

Comment:
See comment above about instanceof -- do people really subclass IntegerType?

Reply:
Also it may be good to say `pyspark.sql.types.IntegerType` for this first one so people know what package it's in.